pgrx/src/spi/cursor.rs

```diff
@@ -67,18 +67,14 @@ pub struct SpiCursor<'client> {
     pub(crate) client: &'client SpiClient,
 }
 
-impl SpiCursor<'_> {
+impl<'client> SpiCursor<'client> {
     /// Fetch up to `count` rows from the cursor, moving forward
     ///
     /// If `fetch` runs off the end of the available rows, an empty [`SpiTupleTable`] is returned.
-    pub fn fetch(&mut self, count: libc::c_long) -> std::result::Result<SpiTupleTable, SpiError> {
-        // SAFETY: no concurrent access
-        unsafe {
-            pg_sys::SPI_tuptable = std::ptr::null_mut();
-        }
+    pub fn fetch(&mut self, count: libc::c_long) -> SpiResult<SpiTupleTable<'client>> {
```

Comment:
I think `'client` would have ended up being used here anyhow, but I wanted it to be explicit.

Note that `SpiCursor::fetch()` doesn't return anything that's owned by itself.  It's returning a `SpiTupleTable` whose lifetime is governed by the active `SpiClient`.

Reply:
I'm feeling a bit suspicious about the lifetimes here: is SpiCursor itself even actually bound to the lifetime of the SpiClient? https://www.postgresql.org/docs/current/spi-spi-cursor-open.html

Not that we cannot or even _should not_ use a shorter lifetime, but it feels unclear here.

Reply:
I haven't quite gotten to thinking about cursor.  It does seem right tho.  

The results of `SpiCursor::fetch()` (the `SpiTupleTable`) are indeed tied to the active `SpiClient`.  

And I think we need to assume the cursor (Portal) is too, unless `SpiCursor::detach_to_name()` is called.  Then the magic happens inside Postgres to let it live longer.

I don't think you can read those docs as "the PortalData pointer returned by SPI_cursor_open can outlive the active SPI connection"

at least that's my understanding.  I have not looked at the Postgres sources around this yet.

Reply:
but `detach_into_name` is just allocating a String?

I don't mind if we handle the name-indexing case specially, mind, instead of using a lifetime binding for it, as it seems to be better for our sanity to handle it that way, I just want to figure out what's going on.